pkg/operator/encryption/controllers/migration_controller.go

```diff
@@ -198,14 +198,19 @@ func (c *migrationController) migrateKeysIfNeededAndRevisionStable() (resetProgr
 
 		if needsMigration(writeSecret, gr) {
 			// storage migration takes a long time so we expose that via a distinct status change
-			if err := c.setProgressing(strings.Title(groupToHumanReadable(gr))+strings.Title(gr.Resource), "migrating resource %s.%s to new write key", groupToHumanReadable(gr), gr.Resource); err != nil {
+			reason, message := strings.Title(groupToHumanReadable(gr))+strings.Title(gr.Resource), fmt.Sprintf("migrating resource %s.%s to new write key", groupToHumanReadable(gr), gr.Resource)
+			if err := c.setProgressing(reason, message); err != nil {
 				return false, err
 			}
+			c.eventRecorder.Eventf("EncryptionMigrationStarted", fmt.Sprintf("Started %s", message))
 
 			if err := c.runStorageMigration(gr); err != nil {
+				c.eventRecorder.Eventf("EncryptionMigrationFailed", fmt.Sprintf("Failed %s", message))
```

Comment:
why we don't want to put the error in the event?

Reply:
good catch